### PR TITLE
A POC that replaces all BigInt to JSBI.BigInt

### PR DESCRIPTION
**Pull request by amoshydra**
## Summary

This is a proof of concept that replaces all reference of BigInt to JSBI

### ~~Technical note:~~
~~rollup config output interop is set to `default` to ensure compatibility between JSBI CJS default export and CJS environment~~ (doesn't seem necessary)

## Related

Related to #100

## How is this tested?

All existing tests pass @ d4789ffcbf546aff6eba6ff32187db52db3a5ef5

### CODE REVIEW

**Comment by jessealama** [comment]
I appreciate the help here. I'm not sure if replacing native BigInts with a JSBI variant of them matches what I have in mind for this package. I intend this package to be a reference implementation of [proposal-decimal](https://github.com/tc39/proposal-decimal/), not necessarily a production-ready package that works in the widest spectrum of JS environments. BigInts are fairly widely available by now (though I understand that they're not available everywhere).

**Comment by jessealama** [comment]
I've merged in your changes that allow us to emit both CJS and ESM -- thanks! Would you like to rebase your work here on top of the latest state?

File: tsconfig.json [hunk 1]
```diff
@@ -27,9 +27,9 @@
         // "moduleDetection": "auto",                        /* Control what method is used to detect module-format JS files. */
 
         /* Modules */
-        "module": "NodeNext" /* Specify what module code is generated. */,
+        "module": "ES2022" /* Specify what module code is generated. */,
         // "rootDir": "./",                                  /* Specify the root folder within your source files. */
-        "moduleResolution": "NodeNext" /* Specify how TypeScript looks up a file from a given module specifier. */,
+        "moduleResolution": "bundler" /* Specify how TypeScript looks up a file from a given module specifier. */,
```
Comment (amoshydra) [inline tsconfig.json]: Rationale for this change: https://github.com/GoogleChromeLabs/jsbi/issues/96#issuecomment-2480828102

File: src/Decimal128.mts [hunk 2]
```diff
@@ -166,10 +168,18 @@ export class Decimal128 {
...
-            data = n;
+            if (n instanceof JSBI) {
+                // Note: when using babel-plugin-transform-jsbi-to-bigint,
+                // the condition above may get transpiled into "typeof n === 'bigint'",
+                // causing the condition above to always be false.
+                // This is fine as we will be hanlding bigint below.
+                data = handleDecimalNotation(n.toString())
+            } else {
+                data = n;
+            }
```
Comment (amoshydra) [inline src/Decimal128.mts:182]: Note to reviewer:

Support type checking for both `bigint` and `JSBI`. Added disclaimer for `babel-plugin-transform-jsbi-to-bigint`


https://github.com/jessealama/decimal128/pull/101/files/#diff-67bdfe7d7bc90f47c617070e310cba7452f1e2e879e487607347f5650c3e07beR171-R182

**Comment by amoshydra** [comment]
re-worked on top of the latest commit of main branch (b218eaa47be2bd48402fa29c0c3e5b5fbca815e3)

force pushed new commit at d4789ffcbf546aff6eba6ff32187db52db3a5ef5

all existing test passed
image

**Review by jessealama — APPROVED** [review]

**Comment by jessealama** [comment]
Thank you!

**Pull request merged**